src/watertap_contrib/reflo/costing/units/multi_effect_crystallizer.py

```diff
@@ -0,0 +1,358 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pyomo.environ as pyo
+from idaes.core.util.misc import StrEnum
+from idaes.core.util.constants import Constants
+from idaes.core.util.exceptions import ConfigurationError
+from watertap.costing.util import register_costing_parameter_block
+from watertap_contrib.reflo.costing.util import make_capital_cost_var
+
+
+class MultiEffectCrystallizerCostType(StrEnum):
+    mass_basis = "mass_basis"
+    volume_basis = "volume_basis"
+
+
+def build_recovered_nacl_cost_param_block(blk):
+
+    blk.cost = pyo.Param(
+        mutable=True,
+        initialize=0,
+        doc="Recovered cost (sale price) of NaCl",
+        units=pyo.units.USD_2020 / pyo.units.kg,
+    )
+
+    costing = blk.parent_block()
+    costing.register_flow_type("NaCl_recovered", blk.cost)
+
+
+def build_steam_cost_param_block(blk):
+
+    blk.cost = pyo.Param(
+        mutable=True,
+        initialize=0.004,
+        doc="Steam cost",
+        units=pyo.units.USD_2018 * pyo.units.meter**-3,
+    )
+
+    costing = blk.parent_block()
+    costing.register_flow_type("steam", blk.cost)
+
+
+def build_multi_effect_crystallizer_cost_param_block(blk):
+
+    # Crystallizer operating cost information from literature
+    blk.fob_unit_cost = pyo.Var(
+        initialize=675000,
+        doc="Forced circulation crystallizer reference free-on-board cost (Woods, 2007)",
+        units=pyo.units.USD_2007,
+    )
+
+    blk.ref_capacity = pyo.Var(
+        initialize=1,
+        doc="Forced circulation crystallizer reference crystal capacity (Woods, 2007)",
+        units=pyo.units.kg / pyo.units.s,
+    )
+
+    blk.ref_exponent = pyo.Var(
+        initialize=0.53,
+        doc="Forced circulation crystallizer cost exponent factor (Woods, 2007)",
+        units=pyo.units.dimensionless,
+    )
+
+    blk.iec_percent = pyo.Var(
+        initialize=1.43,
+        doc="Forced circulation crystallizer installed equipment cost (Diab and Gerogiorgis, 2017)",
+        units=pyo.units.dimensionless,
+    )
+
+    blk.volume_cost = pyo.Var(
+        initialize=16320,
+        doc="Forced circulation crystallizer cost per volume (Yusuf et al., 2019)",
+        units=pyo.units.USD_2007,  ## TODO: Needs confirmation, but data is from Perry apparently
```

Comment:
What is the todo note in reference to? The doc says Yusuf et al., 2019, is the reference, but the comment indicates otherwise. Are we talking about the value itself needing confirmation, the units (year), or something else?

Reply:
I am not sure... I believe this is a direct copy/paste from the WaterTAP main crystallizer costing pkg, but I will have to check.